File: EXTRAS/RapidPrototype_Example/MATLAB/test_script.m
% Define the serial port and parameters
serialPortName = 'COM8';  % Modify this to your specific serial port
baudRate = 115200;

try
    % Create a serialport object
    s = serialport(serialPortName, baudRate);

    while true
        % Generate the message in the specified format
        for value = 45:45:360
            message = sprintf('d%03d', value);
            % Send the message over the serial port
            write(s, message, "char");
            disp(['Sent: ' message]);
            for p = [0, 180]
                message = sprintf('p%03d', p);
                % Send the message over the serial port
                write(s, message, "char");
                disp(['Sent: ' message]);
                pause(2);  % Wait for 2 seconds
            end
        end
    end
catch exception
    close(s);
    error('Error occurred: %s', exception.message);
end

% Close the serial port
close(s);

File: EXTRAS/LCD_I2C_Example/Python/test_script.py
import serial
import time
import matplotlib.pyplot as plt

def send_message(ser, message):
    ser.write(message.encode())
    print(f"Sent: {message}")


def main():
    # Replace 'COM1' with your actual serial port
    serial_port = 'COM8'
    baud_rate = 115200

    ser = serial.Serial(serial_port, baud_rate, timeout=1)

    try:
        while True:
            for x in range(8):
                message = '\001'
                send_message(ser, message)
                time.sleep(0.2)
                message = '\002'
                send_message(ser, message)
                time.sleep(0.2)
            for x in range(8):
                message = '\002'
                send_message(ser, message)
                time.sleep(0.2)
                message = '\001'
                send_message(ser, message)
                time.sleep(0.2)

    except KeyboardInterrupt:
        pass
    finally:
        ser.close()

if __name__ == "__main__":
    main()
